### PR TITLE
AMP-CLIv0.1

### LINKED ISSUE #19: Creation of Software CLI tool
## Description
Build a CLI tool that can be expanded for other AMP developers to use. Currently, should be in charge of building and running the docker container, as well as formatting if possible. For cross-compatibility, preferably made using Python3.

## Sub-tasks
- [x] Research Needed
- [x] Implement docker building and execution
- [x] Implement option to mount current directory in docker image
- [x] Implement function to set up docker
- [x] Optimize docker layout to take advantage of docker cache
- [x] Write in detail help functions
- [ ] User testing
- [x] Refactor it to make the program legible

## Resources
* https://github.com/docker/docker-py
* https://click.palletsprojects.com/en/8.0.x/

### PR DESCRIPTION
**Pull request by alanssitis**
## What is a quick description of the change?
Implementation of AMP-CLI. Should be considered as a beta-release for testing.

## Is this fixing an issue?
Close #19 

## Were any issues created as a result of this change?
#30 and #31

## Check lists (check x in [ ] of list items)
For each of these items, please refer to the [software style guide](https://github.com/Autonomous-Motorsports-Purdue/AMP_ASSv2/wiki/Software-Style-Guide)
- [x] Coding style

## Any additional comments?
None

### CODE REVIEW

**Comment by alanssitis** [comment]
@zghera @Reschivon Could any of you review this PR? I will need some of these docker images in order to work on the CI pipeline for building.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from setuptools import setup
+
+setup(
+    name='AMP-CLI',
+    version='0.1',
+    py_modules=['amp_cli'],
+    install_requires=[
+        'click',
+        'docker',
```
Comment (zghera) [inline setup.py:9]:
```suggestion
        'docker',
        'wheel`,
```
I also had to `pip install wheel` for this to build properly. So we may want to add that here.

**Review by zghera — CHANGES_REQUESTED** [review]
I am not sure why I am unable to run `amp-cli` with either the commands (devel or scratch). I am in fact logged in because I am able to run the hello-world container. See below:

```
(amp-v2) z:~/Documents/AMP_ASSv2(cli-test)$ amp-cli devel
Running amp-devel:frame-desktop...
Error: 404 Client Error for http+docker://localhost/v1.41/images/create?tag=frame-desktop&fromImage=amp-devel: Not Found ("pull access denied for amp-devel, repository does not exist or may require 'docker login': denied: requested access to the resource is denied")
(amp-v2) z:~/Documents/AMP_ASSv2(cli-test)$ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
2db29710123e: Already exists 
Digest: sha256:97a379f4f88575512824f3b352bc03cd75e239179eea0fecc38e597b2209f49a
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

...
```

Maybe I am missing something trivial here as I am not super familiar with docker. Please let me know how I can help to troubleshoot! :)

**Comment by alanssitis** [comment]
Try running it with the build flag!

**Comment by zghera** [comment]
> Try running it with the build flag!

Ahhhhh, I probably should have ran `amp-cli devel --help`, silly me. Is there any way you could catch that exception in your code and advise the user to run with `--help` or `--build` in case someone forgets like me?

**Comment by Reschivon** [comment]
Looks good to me!

**Review by Reschivon — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (alanssitis) [reply setup.py:9]: Gotcha

**Review by zghera — APPROVED** [review]
LGTM!

**Pull request merged**